lib/commands/config/config-apply.ts

```diff
@@ -1,12 +1,19 @@
 export class ConfigApplyCommand implements ICommand {
-	constructor(private $nsCloudServerConfigManager: IServerConfigManager,
-		private $stringParameterBuilder: IStringParameterBuilder) { }
-
 	public allowedParameters: ICommandParameter[] = [this.$stringParameterBuilder.createMandatoryParameter("Configuration name cannot be empty.")];
 
+	public get dashedOptions() {
+		return this.$nsCloudOptionsProvider.dashedOptions;
+	}
+
+	constructor(private $nsCloudOptionsProvider: ICloudOptionsProvider,
+		private $nsCloudServerConfigManager: IServerConfigManager,
+		private $options: ICloudOptions,
+		private $stringParameterBuilder: IStringParameterBuilder) { }
+
 	public async execute(args: string[]): Promise<void> {
 		const configurationName = args[0];
-		this.$nsCloudServerConfigManager.applyConfig(configurationName);
+		const globalServerConfig = { apiVersion: this.$options.apiVersion, serverProto: this.$options.serverProto };
```

Comment:
As a personal preference and definitely not a merge-stopper, I'd suggest marking this as `IServerConfigBase` so that it becomes:
```TypeScript
const globalServerConfig: IServerConfigBase = { apiVersion: this.$options.apiVersion, serverProto: this.$options.serverProto };
```
that way one would have autocomplete when trying to create the static object and in addition compilation would fail should you add any mandatory property to the `IServerConfigBase` interface in the future.